In [13]:
# Imports
import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split,GridSearchCV
from sklearn.preprocessing import StandardScaler
from sklearn.svm import SVC
from sklearn.metrics import confusion_matrix,accuracy_score

In [3]:
df = pd.read_csv('Advertising_data.csv')
df.head()

User ID  Gender   Age  EstimatedSalary  Purchased
0  15624510    Male  19.0          19000.0          0
1  15810944    Male  35.0          20000.0          0
2  15668575  Female  26.0          43000.0          0
3  15603246  Female  27.0          57000.0          0
4  15804002    Male  19.0          76000.0          0

In [4]:
X = df.iloc[:,[2,3]].values
y = df.iloc[:,4].values

In [5]:
X_train, X_test, y_train, y_test = train_test_split( X, y, test_size=0.33, random_state=5)

In [6]:
# Feature Scaling
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)

In [7]:
classifier = SVC(kernel='linear',random_state=0)
classifier.fit(X_train,y_train)

SVC(kernel='linear', random_state=0)

In [8]:
predictions = classifier.predict(X_test)

In [10]:
confusion_matrix(y_test,predictions)

array([[79,  7],
       [13, 33]], dtype=int64)

In [11]:
accuracy_score(y_test,predictions)

0.8484848484848485

In [14]:
# Applying Grid Search to find the best model and the best parameters
parameters = [{'C': [1,10,100,1000], 'kernel' : ['linear']},
            {'C': [1,10,100,1000], 'kernel' : ['rbf'], 'gamma': [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9]}]
grid_search = GridSearchCV(estimator=classifier,
                          param_grid=parameters,
                          scoring='accuracy',
                          cv=10,
                          n_jobs=-1)
grid_search = grid_search.fit(X_train, y_train)

In [15]:
grid_search.best_score_

0.9068376068376068

In [16]:
grid_search.best_params_

{'C': 10, 'gamma': 0.5, 'kernel': 'rbf'}

In [17]:
classifier = SVC(kernel='rbf',gamma=0.5,C=10)
classifier.fit(X_train,y_train)

SVC(C=10, gamma=0.5)

In [18]:
predictions = classifier.predict(X_test)

In [19]:
confusion_matrix(y_test, predictions)

array([[80,  6],
       [ 4, 42]], dtype=int64)

In [20]:
accuracy_score(y_test, predictions)

0.9242424242424242

#### Accuracy before Grid Search = 0.8484848484848485
#### Accuracy after Grid Search = 0.9242424242424242